### PR TITLE
enhanced application name validation - pattern per provider

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Providers will be responsible for setting a valid pattern for application names.

Also removing the specific email error, since it shows up twice - once below the email address, and again at the bottom of the form with the other errors.

cc @duftler

### CODE REVIEW

File: app/scripts/modules/core/application/modal/createApplication.modal.controller.js [hunk 1]
```diff
@@ -118,15 +113,6 @@ module.exports = angular
...
-        return msg
-          .toLowerCase()
-          .indexOf('email') > -1;
-      });
-    }
-
-
     function extractErrorMsg(error) {
       $log.debug('extract error', error);
       assignErrorMsgs();
```
Comment (duftler) [inline app/scripts/modules/core/application/modal/createApplication.modal.controller.js]: Looks like this method was removed.

**Comment by duftler** [comment]
LGTM after comment is addressed.

**Comment by anotherchrisberry** [comment]
Note: in the amended commit, I cleaned up the way we submit the multi-account changes. Instead of sending completely separate tasks and waiting for each one to finish, we are now sending a series of tasks, which cleans up a lot of messy async code.

**Comment by duftler** [comment]
LGTM.

**Pull request merged**